### PR TITLE
ImageEditor: Handle when image fails to load

### PR DESCRIPTION
**Pull request by retrofox**
This PR suggests a solution for this issue #7971.

* When the image can't be loaded ImageEditorCanvas will execute the bound function tied to its `onLoadError` property:

```es6
getImage( url ) {
		const { onLoadError, mimeType } = this.props;

		// ...
		req.onerror = error => onLoadError( error );
		req.send();
```

```es6
<EditCanvas onLoadError={ this.onLoadCanvasError } />
```

* The `onLoadCanvasError()` show an inline error.

~~* The modal dialog will be closed automatically after passing 8 seconds.~~

### Testing

Updated: After merge #7977 we are hiding the `Edit Image` button for private sites to avoid this situation. For this reason I've added a testing flag just to jump this improvement and in this way can test the error handling. Once this PR is ready to merge let's remove this flag

run calypso with:

```cli
> ENABLE_FEATURES=image-editor/force-show-edit-image-button make run
```

1. Create / edit a new post from a private site
2. Add media to the post
3. Start to edit an image (bottom left button)
4. Try to edit the image (top right button)

You should get the inline error

cc @gwwar @obenland @rralian

### CODE REVIEW

**Comment by gwwar** [comment]
If we're closing this automatically in X seconds, we might want to add some feedback that we're doing that. Maybe "We are unable to edit this image. Closing the dialog in X seconds" Let's see if other folks have some better ideas.

File: client/post-editor/media-modal/image-editor/index.jsx [hunk 1]
```diff
@@ -126,18 +139,41 @@ const MediaModalImageEditor = React.createClass( {
 		return ! transfer.types || -1 !== Array.prototype.indexOf.call( transfer.types, 'Files' );
 	},
 
+	onLoadCanvasError() {
+		this.setState( { canvasError: this.translate( 'We are unable to edit this image.' ) } );
+		this.timerId = setTimeout( this.props.onImageEditorCancel, 5000 );
```
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/index.jsx]: Maybe move the 5000, into a const at the top of this file.

File: client/post-editor/media-modal/image-editor/index.jsx [hunk 2]
```diff
@@ -126,18 +139,41 @@ const MediaModalImageEditor = React.createClass( {
...
-				<figure>
-					<div className="editor-media-modal-image-editor__content editor-media-modal__content" >
-						<EditCanvas ref="editCanvas" />
-						<EditToolbar />
-						<EditButtons
-							onCancel={ this.props.onImageEditorCancel }
-							onDone={ this.onDone } />
-					</div>
-				</figure>
+			<div>
```
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/index.jsx]: The outermost div should have the component class name. It looks like we're not matching the naming guidelines at the moment, but let's leave that for another PR.

**Comment by gwwar** [comment]
Left a few suggestions but this works well 👍 

We should be able to 🚢 after the auto-close is removed, or we provide some better feedback for that. Design help is appreciated in finding a good solution for this case.

**Comment by folletto** [comment]
> If we're closing this automatically in X seconds, we might want to add some feedback that we're doing that. 

Agreed. But first:

> The modal dialog will be closed automatically after passing 8 seconds.

Why this behaviour? I'd have expected that message to stay on screen, but maybe I'm missing some piece in my reasoning. :)

**Comment by retrofox** [comment]
> > The modal dialog will be closed automatically after passing 8 seconds.

> Why this behaviour? I'd have expected that message to stay on screen, but maybe I'm missing some piece in my reasoning. :)

No, you aren't missing anything :-D. It was just a suggestion when I was working on this. Let's remove auto-close and give to the user the ability to close the modal using event the cross-close button or the `Cancel` button.

File: client/post-editor/media-modal/image-editor/index.jsx (hunk 2 above)
Comment (retrofox) [inline client/post-editor/media-modal/image-editor/index.jsx]: > It looks like we're not matching the naming guidelines at the moment

Do you mean to `editor-media-modal-image-editor`? It should be named with `image-editor` I guess.

**Comment by retrofox** [comment]
Just FYI I've updated the PR description.

File: client/post-editor/media-modal/detail/detail-item.jsx [hunk 3]
```diff
@@ -48,7 +48,10 @@ module.exports = React.createClass( {
 		const { item, onEdit, site } = this.props;
 
 		// Do not render edit button for private sites
-		if ( site.is_private ) {
+		if (
+				site.is_private &&
+				! config.isEnabled( 'image-editor/force-show-edit-image-button' )
```
Comment (gwwar) [inline client/post-editor/media-modal/detail/detail-item.jsx]: What's this new config value for, testing?

**Comment by gwwar** [comment]
👍 Retested this and code looks good. Let's :shipit:

**Comment by retrofox** [comment]
rebase. waiting test to deploy.
